### PR TITLE
Searching funtionality

### PR DESCRIPTION
**Pull request by SarthakSavasil**

### CODE REVIEW

File: emotunes-commons/src/main/java/com/emotunes/emotunes/dao/UserSongMappingDao.java [hunk 1]
```diff
@@ -53,6 +49,11 @@ public List<SongMetadata> getPaginatedLikedSongs(String userId, String songId, i
 
         return getAndShuffleSongMetaDataList(userSongMappingList);
     }
+    public List<SongMetadata> getSongsDetailsForUserAndSongIds(String userId, List<String> songIdList) {
```
Comment (krishrustagi) [inline emotunes-commons/src/main/java/com/emotunes/emotunes/dao/UserSongMappingDao.java]: rename, 'details' is always redundant

File: emotunes-commons/src/main/java/com/emotunes/emotunes/repository/UserSongMappingRepository.java [hunk 2]
```diff
@@ -59,4 +59,11 @@
             nativeQuery = true
     )
     void updateSongEmotionForUser(String userId, String songId, String correctEmotion);
+
+    @Query(value =
+            "select * from user_song_emotion_mapping"
+                    + " where user_id = ?1 and song_id in ?2",
```
Comment (krishrustagi) [inline emotunes-commons/src/main/java/com/emotunes/emotunes/repository/UserSongMappingRepository.java:65]: got it now!!

**Review by krishrustagi — APPROVED** [review]
LGTM. One change

File: emotunes-commons/src/main/java/com/emotunes/emotunes/dao/UserSongMappingDao.java (hunk 1 above)
Comment (SarthakSavasil) [reply emotunes-commons/src/main/java/com/emotunes/emotunes/dao/UserSongMappingDao.java]: mai songsIds pass kr rha hoon toh song usko point kr rha isiliye with emotion and isiked data ko details refer kiya hai

File: emotunes-commons/src/main/java/com/emotunes/emotunes/dao/UserSongMappingDao.java [hunk 3]
```diff
@@ -76,8 +77,10 @@ public void toggleLike(String userId, String songId) {
         userSongMappingRepository.toggleLike(userId, songId);
     }
 
+
```
Comment (krishrustagi) [inline emotunes-commons/src/main/java/com/emotunes/emotunes/dao/UserSongMappingDao.java]:
```suggestion
```

**Pull request merged**